source/_docs/authentication/multi-factor-auth.markdown

```diff
@@ -14,6 +14,51 @@ footer: true
 This is an advanced feature. If misconfigured, you will not be able to access Home Assistant anymore!
 </p>
 
-Besides the normal authentication providers, it's also possible to configure multi-factor authentication providers. These authentication providers will require the user to solve a second challenge besides just logging in. The idea is that you ask the user for something they know, their username/password, and something they have, like a time-based authentication token from their phone.
+Besides the authentication providers, it's also possible to configure multi-factor authentication modules. These authentication modules will require the user to solve a second challenge besides just logging in. The idea is that you ask the user for something they know, their username/password, and something they have, like a time-based authentication token from their phone.
 
-This feature is currently a work in progress and no configurable multi-factor authentication providers are currently available.
+Multi-factor authentication module can be used mixed-match with authentication providers. After normal authentication provider validation, the login flow will ask user for addional challenge if there are multi-factor authentication modules enabled for this user. If more than one mutli-factor authentication module enabled, user can select one of them during the login.
+
+Multi-factor authentication module has to be enabled for user before it can be used in the login process, user can go to profile page enable it by himself.
+
+## {% linkable_title Configuring mutli-factor authentication modules %}
+
+<p class='note warning'>
+By configuring your own instead of using the default configuration, you take full responsibility for the authentication of the system.
+</p>
+
+Multi-factor authentication modules are configured in your `configuration.yaml` under the `homeassistant:` block:
+
+```yaml
+homeassistant:
+  auth_mfa_modules:
+    - type: totp
+      id: totp_google
+      name: Google Authenticator
```

Comment:
We should keep the examples minimal. Available options can be shown in an advanced example, but only after we explain what all the options do